contracts/validator_manager.v.py

```diff
@@ -198,6 +209,9 @@ def register_notary() -> bool:
 def deregister_notary() -> bool:
     assert self.does_notary_exist[msg.sender] == True
 
+    # Update notary_sample_size
+    self.update_notary_sample_size()
+
     # Delete entry in notary pool
     index_in_notary_pool: int128 = self.notary_registry[msg.sender].pool_index 
```

Comment:
If a notary has deregistered, do we need to `next_period_notary_sample_size - 1` ?

Reply:
At first we decrease `next_period_notary_sample_size ` by one if the notary with index equal to `next_period_notary_sample_size - 1` deregister but then we found out that there is still case like this:
```
At first six notaries are in the pool and then notaries in the middle start to deregister.
[0, 1, 2, 3, 4, 5], sample size=6
[0, 1, _, 3, 4, 5], sample size=6
[0, 1, _, _, 4, 5], sample size=6
[0, 1, _, _, _, 5], sample size=6
[0, 1, _, _, _, _], sample size=5(should be 2)
```

Reply:
Got it! Thanks for the explanations!